modules/apps/address/address-impl/src/main/java/com/liferay/address/internal/search/spi/model/query/contributor/AddressModelPreFilterContributor.java

```diff
@@ -80,7 +88,36 @@ private void _filterByTypeId(
 			return;
 		}
 
-		long[] typeIds = (long[])params.get("typeIds");
+		long[] typeIds = (long[])params.getOrDefault("typeIds", new long[0]);
+		String[] typeNames = (String[])params.get("typeNames");
+
+		if (ArrayUtil.isNotEmpty(typeNames)) {
+			long classNameId = GetterUtil.getLong(
+				searchContext.getAttribute(Field.CLASS_NAME_ID));
+
+			if (classNameId > 0) {
+				ClassName className = _classNameLocalService.fetchByClassNameId(
+					classNameId);
+
+				if (className != null) {
+					List<Long> listTypeIds = new ArrayList<>();
+
+					for (String typeName : typeNames) {
+						ListType listType = _listTypeLocalService.getListType(
+							typeName,
+							className.getClassName() +
+								ListTypeConstants.ADDRESS);
+
+						listTypeIds.add(listType.getListTypeId());
+					}
+
+					typeIds = ArrayUtil.append(
+						typeIds, ArrayUtil.toLongArray(listTypeIds));
+
+					params.put("typeIds", ArrayUtil.unique(typeIds));
```

Comment:
What does this `put` statement do?

Reply:
This is to convert the type names to ids to utilize the indexed `typeId` field for search.

Reply:
Hmm, it doesn't look like the value is ever being pulled off the `params` again. Rather, the `typeIds` variable is being read in the next `if` block.

Reply:
ohh, oops. I forgot I refactored the code and move the read from the params to the top of the method. I just need to set the new value to the `typeIds` var instead of putting it back to `params`.